src/Uno.Extensions.Reactive/Core/ListState.Extensions.cs

```diff
@@ -122,6 +122,114 @@ public static ValueTask AddAsync<T>(this IListState<T> state, T item, Cancellati
+	/// <param name="ct">A token to abort the async add operation.</param>
+	/// <returns></returns>
+	public static ValueTask UpdateItemAsync<T>(this IListState<T> state, T oldItem, T newItem, CancellationToken ct = default)
+		where T : notnull, IKeyEquatable<T>
+		=> state.UpdateDataAsync(
+			itemsOpt => itemsOpt.Map(items =>
+			{
+				var updated = items;
+				foreach (var item in items)
+				{
+					if (item.KeyEquals(oldItem))
+					{
+						updated = items.Replace(item, newItem);
+					}
+				}
+				return updated;
+			}),
+			ct);
+
+
+	/// <summary>
+	/// Updates all items from a list state that match the key of <paramref name="oldItem"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of the items in the list.</typeparam>
+	/// <param name="state">The list state onto which the item should be added.</param>
+	/// <param name="oldItem">The old value of the item.</param>
+	/// <param name="newItem">The new value for the item.</param>
+	/// <param name="ct">A token to abort the async add operation.</param>
+	/// <returns></returns>
+	public static ValueTask UpdateItemAsync<T>(this IListState<T?> state, T oldItem, T? newItem, CancellationToken ct = default)
+		where T : struct, IKeyEquatable<T>
+		=> state.UpdateDataAsync(
+			itemsOpt => itemsOpt.Map(items =>
+			{
+				var updated = items;
+				foreach (var item in items)
+				{
+					if (item?.KeyEquals(oldItem) == true)
+					{
+						updated = items.Replace(item, newItem);
+					}
+				}
+				return updated;
+			}),
+			ct);
+
+
+	/// <summary>
+	/// Updates all items from a list state that match the key of <paramref name="oldItem"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of the items in the list.</typeparam>
+	/// <param name="state">The list state onto which the item should be added.</param>
+	/// <param name="oldItem">The old value of the item.</param>
+	/// <param name="updater">How to update items.</param>
+	/// <param name="ct">A token to abort the async add operation.</param>
+	/// <returns></returns>
+	public static ValueTask UpdateItemAsync<T>(this IListState<T> state, T oldItem, Func<T, T> updater, CancellationToken ct = default)
+		where T : notnull, IKeyEquatable<T>
+		=> state.UpdateDataAsync(
+			itemsOpt => itemsOpt.Map(items =>
+			{
+				var updated = items;
+				foreach (var item in items)
+				{
+					if (item.KeyEquals(oldItem))
+					{
+						updated = items.Replace(item, updater(item));
+					}
+				}
+				return updated;
+			}),
+			ct);
+
+	/// <summary>
+	/// Updates all items from a list state that match the key of <paramref name="oldItem"/>.
+	/// </summary>
+	/// <typeparam name="T">The type of the items in the list.</typeparam>
+	/// <param name="state">The list state onto which the item should be added.</param>
+	/// <param name="oldItem">The old value of the item.</param>
+	/// <param name="updater">How to update items.</param>
+	/// <param name="ct">A token to abort the async add operation.</param>
```

Comment:
nit: the `ct` param comment mentions it is for aborting the **add** operation. Probably just a copy-paste thing but this isn't necessarily an add operation. We could be more general and just mention it as an `async operation`

Reply:
Thanks fixed